Code/Tools/FBuild/FBuildWorker/FBuildWorkerOptions.cpp

```diff
@@ -44,53 +76,20 @@ bool FBuildWorkerOptions::ProcessCommandLine( const AString & commandLine )
     for ( const AString * it=tokens.Begin(); it != end; ++it )
     {
         const AString & token = *it;
-        if ( token.BeginsWith( "-cpus=" ) )
-        {
-            int32_t numCPUs = Env::GetNumProcessors();
-            int32_t num( 0 );
-            if ( sscanf( token.Get() + 6, "%i", &num ) == 1 )
-            {
-                if ( token.EndsWith( '%' ) )
-                {
-                    num = (int32_t)( numCPUs * (float)num / 100.0f );
-                    m_CPUAllocation = Math::Clamp( num, 1, numCPUs );
-                    m_OverrideCPUAllocation = true;
-                    continue;
-                }
-                else if ( num > 0 )
-                {
-                    m_CPUAllocation = Math::Clamp( num, 1, numCPUs );
-                    m_OverrideCPUAllocation = true;
-                    continue;
-                }
-                else if ( num < 0 )
-                {
-                    m_CPUAllocation = Math::Clamp( ( numCPUs + num ), 1, numCPUs );
-                    m_OverrideCPUAllocation = true;
-                    continue;
-                }
-                // problem... fall through
-            }
-            // problem... fall through
-        }
-        else if ( token == "-mode=disabled" )
-        {
-            m_WorkMode = WorkerSettings::DISABLED;
-            m_OverrideWorkMode = true;
-            continue;
-        }
-        else if ( token == "-mode=idle" )
+        if ( token.BeginsWith( "-cpusWhenIdle=" ) )
```

Comment:
Some users launch the FBuildWorker with a script or through other automation. It would be nice if we could extend the command line options while retaining backwards compatibility.
Perhaps we can keep -cpus as is, while adding -cpusWhenIdle?